external-import/sentinel-incidents/src/sentinel_incidents_connector/client_api.py

```diff
@@ -35,18 +36,76 @@ def _get_oauth_token(self) -> str:
         except Exception as e:
             raise ValueError("[ERROR] Failed generating oauth token {" + str(e) + "}")
 
+    def pagination_incidents(self, initial_url: str) -> list:
```

Comment:
I think you could split into several methods with narrower scopes

A query builder

An simple method that know how to retry
=> using HTTPAdapter and Retry from requests.adapters rather than while loop
=> Handle RetryError, HTTPError , Timeout and ConnectionError errors specifically.

A method that orchestrates pagination retrieval